src/transformer.rs

```diff
@@ -9,29 +9,31 @@ use swc_core::ecma::{
 };
 
 use swc_ecma_ast::{
-    ContentTagContent, ExportDefaultExpr, ExprOrSpread, ModuleDecl, ModuleItem, Tpl, TplElement,
+    ContentTagContent, ExportDefaultExpr, ExprOrSpread, Module, ModuleDecl, ModuleItem, Tpl,
+    TplElement,
 };
 
+use swc_ecma_visit::as_folder;
+
 use swc_atoms::Atom;
 
-pub struct TransformVisitor<'a> {
+pub struct Transformer {
     template_identifier: Ident,
-    found_it: Option<&'a mut bool>,
+    pub has_template_tags: bool,
```

Comment:
Inverted the awaked mut pointer passing, and instead make the Transformer expose this field, the caller is free to read it